src/image_transfer.py

```diff
@@ -0,0 +1,286 @@
+import os
+import sys
+import json
+import boto3
+import logging
+
+from invoke import run
+from botocore.exceptions import ClientError
+from datetime import datetime, timezone
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.addHandler(logging.StreamHandler(sys.stdout))
+LOGGER.setLevel(logging.INFO)
+
+from test import test_utils
+
+
+def get_repository_uri(image_uri):
+    """
+    Returns the repository URI in the format <ACCOUNT_ID>.dkr.ecr.<REGION>.amazonaws.com/<REPOSITORY_NAME>
+    """
+    return image_uri.split(":")[0]
+
+
+def pull_image_locally_with_all_its_tags_attached(image_uri):
+    """
+    Pulls the image locally and fetches all its tags and tags the image on the local host as well.
+
+    :param image_uri: str, Image URI
+    :return: List, List of all the tags attached to the image on the ECR
+    """
+    run(f"docker pull {image_uri}", hide=True)
+    image_region = test_utils.get_region_from_image_uri(image_uri=image_uri)
+    ecr_client = boto3.client("ecr", region_name=image_region)
+    image_repo = image_uri.split(":")[0]
+    tag_list = test_utils.get_all_the_tags_of_an_image_from_ecr(
+        ecr_client=ecr_client, image_uri=image_uri
+    )
+    for tag in tag_list:
+        run(f"docker tag {image_uri} {image_repo}:{tag}", hide=True)
+    return tag_list
+
+
+def is_latest_benchmark_tested_beta_image_an_autopatch_image_itself(beta_image_uri):
+    """
+    Checks if the latest benchmark_tested_beta_image is an autopatch image itself. It pulls the tags
+    attached to the beta_benchmark_tested_image and checks for the presence of "-autopatch" in any of
+    the tags.
+
+    :param beta_image_uri: str, Image URI of beta benchmark tested image
+    :return: boolean, True if beta benchmark image is an autopatch image itself. False otherwise.
+    """
+    image_region = test_utils.get_region_from_image_uri(image_uri=beta_image_uri)
+    ecr_client = boto3.client("ecr", region_name=image_region)
+    tag_list = test_utils.get_all_the_tags_of_an_image_from_ecr(
+        ecr_client=ecr_client, image_uri=beta_image_uri
+    )
+    return any(["-autopatch" in tag for tag in tag_list])
+
+
+def get_push_time_of_image_from_ecr(image_uri):
+    """
+    This method uses the ERC boto3 client to get the push time of an image.
+
+    :param image_uri: str, Image URI
+    :return: datetime.datetime Object, Time of the Push
+    """
+    image_region = test_utils.get_region_from_image_uri(image_uri=image_uri)
+    ecr_client = boto3.client("ecr", region_name=image_region)
+    return test_utils.get_image_push_time_from_ecr(ecr_client=ecr_client, image_uri=image_uri)
```

Comment:
Why not just use this function instead of creating a wrapper for it here?

Reply:
That's a nice suggestion. I have removed the wrapper function altogether. Thanks!